### PR TITLE
Update common libraries to use PropagatedClaims

### PR DESCRIPTION
**Pull request by sbryzak**
This is a companion PR to https://github.com/codeready-toolchain/host-operator/pull/943 which refactors the MasterUserRecordController to only use the user's SSO claim values from PropagatedClaims and nowhere else.

### CODE REVIEW

**Review by alexeykazakov — APPROVED** [review]

File: pkg/test/masteruserrecord/masteruserrecord.go [hunk 1]
```diff
@@ -93,7 +88,7 @@ func ModifyUaInMur(mur *toolchainv1alpha1.MasterUserRecord, targetCluster string
 
 func UserID(userID string) MurModifier {
 	return func(mur *toolchainv1alpha1.MasterUserRecord) error {
-		mur.Spec.UserID = userID
+		mur.Spec.PropagatedClaims.UserID = userID
 		return nil
```
[the remark this answers is not in the document]
Comment (sbryzak) [reply pkg/test/masteruserrecord/masteruserrecord.go:92]: Host operator PR is already here: https://github.com/codeready-toolchain/host-operator/pull/943

I've created two new PRs as requested, I'm not sure what value these have though:

Registration service: https://github.com/codeready-toolchain/registration-service/pull/380
Member operator: https://github.com/codeready-toolchain/member-operator/pull/512

I've also added a new function for setting the Sub

File: pkg/test/masteruserrecord/masteruserrecord_assertion.go [hunk 2]
```diff
@@ -202,13 +202,6 @@ func (a *MasterUserRecordAssertion) HasAnnotationWithValue(key, value string) *M
 	return a
 }
 
-func (a *MasterUserRecordAssertion) HasOriginalSub(sub string) *MasterUserRecordAssertion {
-	err := a.loadMasterUserRecord()
-	require.NoError(a.t, err)
-	assert.Equal(a.t, sub, a.mur.Spec.OriginalSub)
-	return a
-}
-
```
[the remark this answers is not in the document]
Comment (sbryzak) [reply pkg/test/masteruserrecord/masteruserrecord_assertion.go:211]: @MatousJobanek 

> the description still contains "WIP" so not sure if it's already ready for review or not.

> 
> Anyway, I got a bit lost in the changes you have made and you are planning to do. I'm also not sure when you want to drop the fields from the CRDs.
> 
> Apart from that, you missed a few cases:
> 
>     the usersignup.go file still uses the UserSignupUserEmailAnnotationKey
>     the same for notification_builder.go
>     plus there is also one in the useraccount_test.go 
> 

The PR is indeed ready for review.  Fields won't be dropped from the CRDs until everything has been refactored to only use the PropagatedClaims.

The cases mentioned in usersignup.go and notification_builder.go are not missed, they are not within the scope of this PR and will be updated when the UserSignupController is refactored (these changes are limited to only the MasterUserRecordController).

I've updated useraccount_test.go to perform an equality check of the entire PropagatedClaims instead.

File: pkg/test/masteruserrecord/masteruserrecord.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (MatousJobanek) [reply pkg/test/masteruserrecord/masteruserrecord.go:92]: > I've created two new PRs as requested, I'm not sure what value these have though:

well the reason is that you are changing logic of functions that are (can be) used in these repositories, so it's good to make sure that the repos are compatible with the changes.

> I've also added a new function for setting the Sub

I don't see anything like that in the PR, but still, do you want to keep the changes of the `UserID` function?

File: pkg/test/masteruserrecord/masteruserrecord_assertion.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (MatousJobanek) [reply pkg/test/masteruserrecord/masteruserrecord_assertion.go:211]: cool, thanks for the explanation :+1: I was missing the description in the PR - what the PR is changing and what are the next steps.

> I've updated useraccount_test.go to perform an equality check of the entire PropagatedClaims instead.

I don't see any changes in the file. Apart from that, it's still using the UserId from spec
https://github.com/codeready-toolchain/toolchain-common/blob/6145201176e8055aba0c85477db5bbfbd889997a/pkg/test/useraccount/useraccount_test.go#L19

**Review by MatousJobanek — APPROVED** [review]
Thanks for addressing my comments :+1:

**Pull request merged**